dev_steps.md

```diff
@@ -0,0 +1,53 @@
+# Basic development steps
+
+For developers who want to contribute or modify PECOS:
+
+1. Make sure you have [Python](https://www.python.org/downloads/) and [Rust](https://www.rust-lang.org/tools/install) installed for you system (although you can get away with developing in one or the other).
+
+2. Clone the repository:
+   ```sh
+   git clone https://github.com/PECOS-packages/PECOS.git
+   cd PECOS
+   ```
+
+3. Set up the development environment either using the `Makefile` (Note: for Windows to use the `Makefile` you may need to use a shell that has access to Linux commands such as utilizing [git bash](https://gitforwindows.org/)):
+   ```sh
+   make venv
+   ```
+    or manually following [these steps](venv_setup.md).
+
+    When developing in the development environment, be sure to activate the venv:
+
+    On Linux/Mac:
+    ```sh
+    source .venv/bin/activate
+    ```
+
+    On Windows:
+    ```sh
+    .\venv\Scripts\activate
+    ```
+
+4. Build the project in editable mode
+    ```sh
+   make build
```

Comment:
Failed for me with:
```
INFO: pip is looking at multiple versions of quantum-pecos[simulators] to determine which version is compatible with other requirements. This could take a while.
ERROR: Could not find a version that satisfies the requirement pyquest>=0.0.1; extra == "simulators" (from quantum-pecos[simulators]) (from versions: none)
ERROR: No matching distribution found for pyquest>=0.0.1; extra == "simulators"
```

I believe this was because I was on Python 3.13 at the time. Worked after switching to 3.12. Perhaps pyproject.toml should exclude 3.13 if it's not supported.

Reply:
Hmmm, pyquest isn't a necessary component. Sounds like maybe I need to adjust `make build` or the `pyproject.toml` and re-evaluate 3.13. 

Good catch. Thanks!